vizro-core/tests/unit/vizro/actions/test_parameter_action.py

```diff
@@ -329,3 +390,63 @@ def test_one_parameter_per_target_multiple_attributes(
         expected = {"box_chart": target_box_parameter_y_and_x}
 
         assert result == expected
+
+    @pytest.mark.usefixtures("managers_one_page_two_graphs_with_dynamic_data")
+    @pytest.mark.parametrize(
+        "ctx_parameter_data_frame_argument, "
+        "target_scatter_parameter_data_frame_first_n_last_n, "
+        "target_box_parameter_data_frame_first_n_last_n",
+        [
+            (
+                (["scatter_chart", "box_chart"], {"first_n": 50}),
+                {"first_n": 50},
+                {"first_n": 50},
+            ),
+            (
+                (["scatter_chart", "box_chart"], {"first_n": 50, "last_n": 50}),
+                {"first_n": 50, "last_n": 50},
+                {"first_n": 50, "last_n": 50},
+            ),
+        ],
+        indirect=[
+            "ctx_parameter_data_frame_argument",
+            "target_scatter_parameter_data_frame_first_n_last_n",
+            "target_box_parameter_data_frame_first_n_last_n",
+        ],
+    )
+    def test_data_frame_parameters_multiple_targets(
```

Comment:
Does it make sense to additionally test combinations where only part of the screen gets targeted? That is the only other outstanding case I can think of